httpcore5/src/main/java/org/apache/hc/core5/net/Host.java

```diff
@@ -72,7 +73,10 @@ static Host parse(final CharSequence s, final Tokenizer.Cursor cursor) throws UR
                 throw URISupport.createException(s, cursor, "Expected an IPv6 address");
             }
         } else {
-            hostName = tokenizer.parseContent(s, cursor, URISupport.PORT_SEPARATORS);
+            // If the hostName contains non-ASCII characters, IDNA processing is required
+            hostName = TextUtils.isAllASCII(s) ?
+                    tokenizer.parseContent(s, cursor, URISupport.PORT_SEPARATORS) :
+                    IDN.toASCII(tokenizer.parseContent(s, cursor, URISupport.PORT_SEPARATORS));
```

Comment:
java.net.IDN is implementing a deprecated version of IDNA (IDNA 2003). The latest version, IDNA 2008, can be supported using the ICU4j library